src/components/molecules/ProfilePictureInput/ProfilePictureInput.tsx

```diff
@@ -107,21 +124,37 @@ export const ProfilePictureInput: React.FunctionComponent<ProfilePictureInputPro
     ? customAvatars
     : DEFAULT_AVATARS;
 
-  const avatarImages = useMemo(() => {
-    return defaultAvatars.map((avatar, index) => (
+  const getRenderedAvatarImageComponent = useCallback(
+    (avatarSrc: string, key: string | number) => (
       <div
-        key={`${avatar}-${index}`}
+        key={`${avatarSrc}-${key}`}
         className="profile-picture-preview-container"
-        onClick={() => uploadDefaultAvatar(avatar)}
+        onClick={() => uploadDefaultAvatar(avatarSrc)}
       >
         <img
-          src={avatar}
+          src={avatarSrc}
           className="profile-icon profile-picture-preview"
-          alt={`default avatar ${index}`}
+          alt={`default avatar ${key}`}
         />
       </div>
-    ));
-  }, [defaultAvatars, uploadDefaultAvatar]);
+    ),
+    [uploadDefaultAvatar]
+  );
+
+  const avatarImages = useMemo(() => {
+    const renderedAvatarImages = defaultAvatars.map(
+      getRenderedAvatarImageComponent
+    );
```

Comment:
I still don't see why we would bother keeping `getRenderedAvatarImageComponent` instead of just inlining it within `avatarImages`.

`defaultAvatars` is a `string[]`

`githubImageSrc` is a `string | undefined`

So just do something like:

```typescript
[githubAvatarImage, ...defaultAvatars].filter(isDefined).map((avatarSrc) =>
  <div
    key={avatarSrc}
    // ..snip..
```

Reply:
Done in https://github.com/sparkletown/sparkle/pull/1667/commits/36f9ee2f4cfc1f5342c9ef7c0843ba5452393d9f

Reply:
Why did you use the `index` as part of the `key`?

Reply:
@0xdevalias legacy logic